src/raftstore/store/peer_storage.rs

```diff
@@ -503,6 +503,18 @@ impl PeerStorage {
         Ok(())
     }
 
+    /// Delete all data that is not covered by `new_region`.
+    fn clear_extra_data(&self, new_region: &metapb::Region) -> Result<()> {
+        let timer = Instant::now();
+        let (old_start_key, old_end_key) = (enc_start_key(self.get_region()),
+                                            enc_end_key(self.get_region()));
+        let (new_start_key, new_end_key) = (enc_start_key(new_region), enc_end_key(new_region));
+        try!(delete_all_in_range(&self.engine, &old_start_key, &new_start_key));
```

Comment:
What would happened if met the cross interval:
new_start_key.......old_start_key......new_end_key.....old_end_key
or 
new_start_key.......old_start_key.....old_end_key......new_end_key